packages/s3-store/index.ts

```diff
@@ -235,31 +238,36 @@ export class S3Store extends DataStore {
       .on('chunkFinished', ({path, size: partSize}) => {
         pendingChunkFilepath = null
 
+        const partNumber = currentPartNumber++
+        const chunkNumber = currentChunkNumber++
+
+        offset += partSize
+
+        const isFirstChunk = chunkNumber === 0
+        const isFinalPart = size === offset
+
         // eslint-disable-next-line no-async-promise-executor
         const deferred = new Promise<void>(async (resolve, reject) => {
           try {
-            const partNumber = currentPartNumber++
             const incompletePartId = this.partKey(metadata.file.id, true)
-            // If we received a chunk under the minimum part size in a previous iteration,
-            // we used a regular S3 upload to save it in the bucket. We try to get the incomplete part here.
-            const incompletePart = await this.getIncompletePart(incompletePartId)
-            const isFinalChunk = size === offset + partSize
-
-            if (incompletePart) {
-              // We found an incomplete part, prepend it to the chunk on disk we were about to upload,
-              // and delete the incomplete part from the bucket. This can be done in parallel.
-              await Promise.all([
-                this.prependIncompletePart(path, incompletePart),
-                this.deleteIncompletePart(incompletePartId),
-              ])
+            if (isFirstChunk) {
```

Comment:
why do we only check for an incomplete part if it's the first chunk?

Reply:
Only first chunk of each PATCH request can prepend incomplete part from previous request.

Reply:
I've used different terms to disambiguate. `chunk` is a part of a **current** request , while `part` is part of S3 (multipart) upload.